packages/@aws-cdk/aws-lambda-python/lib/bundling.ts

```diff
@@ -16,137 +17,89 @@ export const BUNDLER_DEPENDENCIES_CACHE = '/var/dependencies';
-   * directory will cause the asset to rebuild. This means, for example, that in
-   * order to pick up a new dependency version, a change must be made to the
-   * source tree. Ideally, this can be implemented by including a dependency
-   * lockfile in your source tree or using fixed dependencies.
    *
-   * If the asset hash is set to `OUTPUT`, the hash is calculated after
-   * bundling. This means that any change in the output will cause the asset to
-   * be invalidated and uploaded. Bear in mind that `pip` adds timestamps to
-   * dependencies it installs, which implies that in this mode Python bundles
-   * will _always_ get rebuild and uploaded. Normally this is an anti-pattern
-   * since build
-   *
-   * @default AssetHashType.SOURCE By default, hash is calculated based on the
-   * contents of the source directory. If `assetHash` is also specified, the
-   * default is `CUSTOM`. This means that only updates to the source will cause
-   * the asset to rebuild.
+   * @default Architecture.X86_64
    */
-  readonly assetHashType?: cdk.AssetHashType;
-
-  /**
-   * Specify a custom hash for this asset. If `assetHashType` is set it must
-   * be set to `AssetHashType.CUSTOM`. For consistency, this custom hash will
-   * be SHA256 hashed and encoded as hex. The resulting hash will be the asset
-   * hash.
-   *
-   * NOTE: the hash is used in order to identify a specific revision of the asset, and
-   * used for optimizing and caching deployment activities related to this asset such as
-   * packaging, uploading to Amazon S3, etc. If you chose to customize the hash, you will
-   * need to make sure it is updated every time the asset changes, or otherwise it is
-   * possible that some deployments will not be invalidated.
-   *
-   * @default - based on `assetHashType`
-   */
-  readonly assetHash?: string;
+  readonly architecture?: Architecture;
 }
 
 /**
  * Produce bundled Lambda asset code
  */
-export function bundle(options: BundlingOptions): lambda.Code {
-  const { entry, runtime, architecture, outputPathSuffix } = options;
-
-  const stagedir = cdk.FileSystem.mkdtemp('python-bundling-');
-  const hasDeps = stageDependencies(entry, stagedir);
-
-  const depsCommand = chain([
-    hasDeps ? `rsync -r ${BUNDLER_DEPENDENCIES_CACHE}/. ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}/${outputPathSuffix}` : '',
-    `rsync -r . ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}/${outputPathSuffix}`,
-  ]);
-
-  // Determine which dockerfile to use. When dependencies are present, we use a
-  // Dockerfile that can create a cacheable layer. We can't use this Dockerfile
-  // if there aren't dependencies or the Dockerfile will complain about missing
-  // sources.
-  const dockerfile = hasDeps
-    ? 'Dockerfile.dependencies'
-    : 'Dockerfile';
-
-  // copy Dockerfile to workdir
-  fs.copyFileSync(path.join(__dirname, dockerfile), path.join(stagedir, dockerfile));
+export class Bundling implements CdkBundlingOptions {
```

Comment:
This is simpler and aligns with what exists for `GoFunction` and `NodeJsFunction`.